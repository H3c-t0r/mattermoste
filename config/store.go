// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See License.txt for license information.

package config

import (
	"github.com/pkg/errors"
	"strings"

	"github.com/mattermost/mattermost-server/model"
)

// Listener is a callback function invoked when the configuration changes.
type Listener func(oldConfig *model.Config, newConfig *model.Config)

// Store abstracts the act of getting and setting the configuration.
type Store interface {
	// Get fetches the current, cached configuration.
	Get() *model.Config

	// GetEnvironmentOverrides fetches the configuration fields overridden by environment variables.
	GetEnvironmentOverrides() map[string]interface{}

	// Set replaces the current configuration in its entirety and updates the backing store.
	Set(*model.Config) (*model.Config, error)

	// Load updates the current configuration from the backing store, possibly initializing.
	Load() (err error)

	// AddListener adds a callback function to invoke when the configuration is modified.
	AddListener(listener Listener) string

	// RemoveListener removes a callback function using an id returned from AddListener.
	RemoveListener(id string)

	// GetFile fetches the contents of a previously persisted configuration file.
	// If no such file exists, an empty byte array will be returned without error.
	GetFile(name string) ([]byte, error)

	// SetFile sets or replaces the contents of a configuration file.
	SetFile(name string, data []byte) error

	// HasFile returns true if the given file was previously persisted.
	HasFile(name string) (bool, error)

	// RemoveFile removes a previously persisted configuration file.
	RemoveFile(name string) error

	// String describes the backing store for the config.
	String() string

	// Close cleans up resources associated with the store.
	Close() error
}

// NewStore creates a database or file store given a data source name by which to connect.
func NewStore(dsn string, watch bool) (Store, error) {
	if strings.HasPrefix(dsn, "mysql://") || strings.HasPrefix(dsn, "postgres://") {
		return NewDatabaseStore(dsn)
	}

	return NewFileStore(dsn, watch)
}

func MigrateConfig(sourceStore, destinationStore Store) error {
	// Copy config from source to destination
	sourceStoreConfig := sourceStore.Get()

	_, err := destinationStore.Set(sourceStoreConfig)
	if err != nil {
		return errors.Wrap(err, "failed to migrate config")
	}

	//get the certificate files from sourceStore
	idpCertificateFile, err := sourceStore.GetFile(*sourceStoreConfig.SamlSettings.IdpCertificateFile)
	if err != nil {
		return errors.Wrap(err, "failed to migrate config")
	}

	publicCertificateFile, err := sourceStore.GetFile(*sourceStoreConfig.SamlSettings.PublicCertificateFile)
	if err != nil {
		return errors.Wrap(err, "failed to migrate config")
	}

	privateKeyFile, err := sourceStore.GetFile(*sourceStoreConfig.SamlSettings.PrivateKeyFile)
	if err != nil {
		return errors.Wrap(err, "failed to migrate config")
	}

	//copy files to destinationStore
	err = destinationStore.SetFile(*sourceStoreConfig.SamlSettings.IdpCertificateFile, idpCertificateFile)
	if err != nil {
		return errors.Wrap(err, "failed to migrate config")
	}

	err = destinationStore.SetFile(*sourceStoreConfig.SamlSettings.PublicCertificateFile, publicCertificateFile)
	if err != nil {
		return errors.Wrap(err, "failed to migrate config")
	}

	err = destinationStore.SetFile(*sourceStoreConfig.SamlSettings.PrivateKeyFile, privateKeyFile)
	if err != nil {
		return errors.Wrap(err, "failed to migrate config")
	}

	return nil
}
