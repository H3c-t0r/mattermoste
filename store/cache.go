package store

import (
	"time"
)

type Cache interface {
	Purge()
	Add(key, value interface{})
	AddWithDefaultExpires(key, value interface{})
	AddWithExpiresInSecs(key, value interface{}, expireAtSecs int64)
	Get(key interface{}) (value interface{}, ok bool)
	GetOrAdd(key, value interface{}, ttl time.Duration) (actual interface{}, loaded bool)
	Remove(key interface{})
	Keys() []interface{}
	Len() int
	Name() string
	GetInvalidateClusterEvent() string
}
