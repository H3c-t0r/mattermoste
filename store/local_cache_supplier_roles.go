// Copyright (c) 2017-present Mattermost, Inc. All Rights Reserved.
// See License.txt for license information.

package store

import (
	"context"

	l4g "github.com/alecthomas/log4go"
	"github.com/mattermost/mattermost-server/model"
)

func (s *LocalCacheSupplier) handleClusterInvalidateRole(msg *model.ClusterMessage) {
	if msg.Data == CLEAR_CACHE_MESSAGE_DATA {
		s.roleCache.Purge()
	} else {
		s.roleCache.Remove(msg.Data)
	}
}

func (s *LocalCacheSupplier) RoleSave(ctx context.Context, role *model.Role, hints ...LayeredStoreHint) *LayeredStoreSupplierResult {
	if len(role.Id) != 0 {
		s.doInvalidateCacheCluster(s.roleCache, role.Name)
	}
	return s.Next().RoleSave(ctx, role, hints...)
}

func (s *LocalCacheSupplier) RoleGet(ctx context.Context, roleId string, hints ...LayeredStoreHint) *LayeredStoreSupplierResult {
	// TODO: No caching. Remove this method altogether from the store.
	return s.Next().RoleGet(ctx, roleId, hints...)
}

func (s *LocalCacheSupplier) RoleGetByName(ctx context.Context, name string, hints ...LayeredStoreHint) *LayeredStoreSupplierResult {
	if result := s.doStandardReadCache(ctx, s.reactionCache, name, hints...); result != nil {
		return result
	}

	result := s.Next().RoleGetByName(ctx, name, hints...)

	s.doStandardAddToCache(ctx, s.reactionCache, name, result, hints...)

	return result
}

func (s *LocalCacheSupplier) RoleGetByNames(ctx context.Context, roleNames []string, hints ...LayeredStoreHint) *LayeredStoreSupplierResult {
	var foundRoles model.Roles

	for _, roleName := range roleNames {
		if result := s.doStandardReadCache(ctx, s.reactionCache, roleName, hints...); result != nil {
			foundRoles = append(foundRoles, result.Data.(*model.Role))
		}
	}

	var rolesToQuery []string
	for _, roleName := range roleNames {
		found := false
		for _, role := range foundRoles {
			if roleName == role.Name {
				l4g.Debug("Found a role in the cache")
				found = true
				break
			}
		}
		if !found {
			rolesToQuery = append(rolesToQuery, roleName)
		}
	}

	result := s.Next().RoleGetByNames(ctx, rolesToQuery, hints...)

	if result.Data != nil {
		rolesFound := result.Data.(model.Roles)
		for _, role := range rolesFound {
			res := NewSupplierResult()
			res.Data = role
			s.doStandardAddToCache(ctx, s.roleCache, role.Name, res, hints...)
		}
		result.Data = append(foundRoles, result.Data.(model.Roles)...)
	}

	return result
}
