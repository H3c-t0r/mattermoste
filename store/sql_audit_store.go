// Copyright (c) 2015 Mattermost, Inc. All Rights Reserved.
// See License.txt for license information.

package store

import (
	"github.com/mattermost/platform/model"
	goi18n "github.com/nicksnyder/go-i18n/i18n"
)

type SqlAuditStore struct {
	*SqlStore
}

func NewSqlAuditStore(sqlStore *SqlStore) AuditStore {
	s := &SqlAuditStore{sqlStore}

	for _, db := range sqlStore.GetAllConns() {
		table := db.AddTableWithName(model.Audit{}, "Audits").SetKeys(false, "Id")
		table.ColMap("Id").SetMaxSize(26)
		table.ColMap("UserId").SetMaxSize(26)
		table.ColMap("Action").SetMaxSize(64)
		table.ColMap("ExtraInfo").SetMaxSize(128)
		table.ColMap("IpAddress").SetMaxSize(64)
		table.ColMap("SessionId").SetMaxSize(26)
	}

	return s
}

func (s SqlAuditStore) UpgradeSchemaIfNeeded() {
}

func (s SqlAuditStore) CreateIndexesIfNotExists(T goi18n.TranslateFunc) {
	s.CreateIndexIfNotExists("idx_audits_user_id", "Audits", "UserId", T)
}

func (s SqlAuditStore) Save(audit *model.Audit, T goi18n.TranslateFunc) StoreChannel {

	storeChannel := make(StoreChannel)

	go func() {
		result := StoreResult{}

		audit.Id = model.NewId()
		audit.CreateAt = model.GetMillis()

		if err := s.GetMaster().Insert(audit); err != nil {
			result.Err = model.NewAppError("SqlAuditStore.Save",
				T("We encounted an error saving the audit"), "user_id="+
					audit.UserId+" action="+audit.Action)
		}

		storeChannel <- result
		close(storeChannel)
	}()

	return storeChannel
}

func (s SqlAuditStore) Get(user_id string, limit int, T goi18n.TranslateFunc) StoreChannel {

	storeChannel := make(StoreChannel)

	go func() {
		result := StoreResult{}

		if limit > 1000 {
			limit = 1000
			result.Err = model.NewAppError("SqlAuditStore.Get", T("Limit exceeded for paging"), "user_id="+user_id)
			storeChannel <- result
			close(storeChannel)
			return
		}

		var audits model.Audits
		if _, err := s.GetReplica().Select(&audits, "SELECT * FROM Audits WHERE UserId = :user_id ORDER BY CreateAt DESC LIMIT :limit",
			map[string]interface{}{"user_id": user_id, "limit": limit}); err != nil {
			result.Err = model.NewAppError("SqlAuditStore.Get", T("We encountered an error finding the audits"), "user_id="+user_id)
		} else {
			result.Data = audits
		}

		storeChannel <- result
		close(storeChannel)
	}()

	return storeChannel
}

func (s SqlAuditStore) PermanentDeleteByUser(userId string, T goi18n.TranslateFunc) StoreChannel {

	storeChannel := make(StoreChannel)

	go func() {
		result := StoreResult{}

		if _, err := s.GetMaster().Exec("DELETE FROM Audits WHERE UserId = :userId",
			map[string]interface{}{"userId": userId}); err != nil {
			result.Err = model.NewAppError("SqlAuditStore.Delete", T("We encountered an error deleting the audits"), "user_id="+userId)
		}

		storeChannel <- result
		close(storeChannel)
	}()

	return storeChannel
}
