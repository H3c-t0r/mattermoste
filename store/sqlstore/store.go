// Copyright (c) 2016-present Mattermost, Inc. All Rights Reserved.
// See License.txt for license information.

// Code generated by enterprise/einterfaces_gen.go; DO NOT EDIT.

package sqlstore

import (
	sq "github.com/Masterminds/squirrel"
	_ "github.com/go-sql-driver/mysql"
	_ "github.com/lib/pq"
	"github.com/mattermost/gorp"
	"github.com/mattermost/mattermost-server/store"
)

type SqlStore interface {
	SetChainNext(next store.LayeredStoreSupplier)
	Next() store.LayeredStoreSupplier
	DriverName() string
	GetCurrentSchemaVersion() string
	GetMaster() *gorp.DbMap
	GetSearchReplica() *gorp.DbMap
	GetReplica() *gorp.DbMap
	TotalMasterDbConnections() int
	TotalReadDbConnections() int
	TotalSearchDbConnections() int
	MarkSystemRanUnitTests()
	DoesTableExist(tableName string) bool
	DoesColumnExist(tableName string, columnName string) bool
	DoesTriggerExist(triggerName string) bool
	CreateColumnIfNotExists(tableName string, columnName string, mySqlColType string, postgresColType string, defaultValue string) bool
	CreateColumnIfNotExistsNoDefault(tableName string, columnName string, mySqlColType string, postgresColType string) bool
	RemoveColumnIfExists(tableName string, columnName string) bool
	RemoveTableIfExists(tableName string) bool
	RenameColumnIfExists(tableName string, oldColumnName string, newColumnName string, colType string) bool
	GetMaxLengthOfColumnIfExists(tableName string, columnName string) string
	AlterColumnTypeIfExists(tableName string, columnName string, mySqlColType string, postgresColType string) bool
	AlterColumnDefaultIfExists(tableName string, columnName string, mySqlColDefault *string, postgresColDefault *string) bool
	CreateUniqueIndexIfNotExists(indexName string, tableName string, columnName string) bool
	CreateIndexIfNotExists(indexName string, tableName string, columnName string) bool
	CreateCompositeIndexIfNotExists(indexName string, tableName string, columnNames []string) bool
	CreateFullTextIndexIfNotExists(indexName string, tableName string, columnName string) bool
	RemoveIndexIfExists(indexName string, tableName string) bool
	GetAllConns() []*gorp.DbMap
	Close()
	LockToMaster()
	UnlockFromMaster()
	Team() store.TeamStore
	Channel() store.ChannelStore
	Post() store.PostStore
	User() store.UserStore
	Bot() store.BotStore
	Session() store.SessionStore
	Audit() store.AuditStore
	ClusterDiscovery() store.ClusterDiscoveryStore
	Compliance() store.ComplianceStore
	OAuth() store.OAuthStore
	System() store.SystemStore
	Webhook() store.WebhookStore
	Command() store.CommandStore
	CommandWebhook() store.CommandWebhookStore
	Preference() store.PreferenceStore
	License() store.LicenseStore
	Token() store.TokenStore
	Emoji() store.EmojiStore
	Status() store.StatusStore
	FileInfo() store.FileInfoStore
	Reaction() store.ReactionStore
	Job() store.JobStore
	UserAccessToken() store.UserAccessTokenStore
	ChannelMemberHistory() store.ChannelMemberHistoryStore
	Plugin() store.PluginStore
	Role() store.RoleStore
	TermsOfService() store.TermsOfServiceStore
	UserTermsOfService() store.UserTermsOfServiceStore
	Scheme() store.SchemeStore
	Group() store.GroupStore
	LinkMetadata() store.LinkMetadataStore
	DropAllTables()
	GetQueryBuilder() sq.StatementBuilderType
}
