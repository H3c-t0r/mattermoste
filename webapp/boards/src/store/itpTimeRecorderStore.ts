// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import {PayloadAction, createSlice} from '@reduxjs/toolkit'

import {RootState} from './index'

const storedValue = localStorage.getItem('ongoingTask');

const triggerSlice = createSlice({
    
    name: 'itpTimeRecorder',
    initialState: { isTriggered: false },
    reducers: {
        setTrigger: (state, action: PayloadAction<boolean>) => {
            localStorage.setItem('ongoingTask', String(action.payload))
            console.log(action.payload);
            state.isTriggered = action.payload;
            
          },
    },
  });
  

export const {reducer} = triggerSlice
export const { setTrigger } = triggerSlice.actions;

export const getTrigger = (state: RootState): boolean => state.itpTimeRecorder.isTriggered

