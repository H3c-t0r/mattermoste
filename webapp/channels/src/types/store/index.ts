// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import type {PluginsState} from './plugins';
import type {ViewsState} from './views';
import type {GlobalState as BaseGlobalState} from '@mattermost/types/store';

export type DraggingState = {
    state?: string;
    type?: string;
    id?: string;
}

export type GlobalState = BaseGlobalState & {
    plugins: PluginsState;
    storage: {
        storage: Record<string, any>;
        initialized: boolean;
    };
    views: ViewsState;
};
