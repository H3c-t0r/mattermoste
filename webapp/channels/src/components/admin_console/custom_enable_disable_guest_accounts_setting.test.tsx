// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import {render, fireEvent, screen} from '@testing-library/react';
import React from 'react';
import {FormattedMessage, IntlProvider} from 'react-intl';
import {Provider} from 'react-redux';

import store from 'stores/redux_store';

import CustomEnableDisableGuestAccountsSetting from './custom_enable_disable_guest_accounts_setting';

describe('components/AdminConsole/CustomEnableDisableGuestAccountsSetting', () => {
    const baseProps = {
        id: 'MySetting',
        value: false,
        onChange: jest.fn(),
        cancelSubmit: jest.fn(),
        disabled: false,
        setByEnv: false,
        showConfirm: false,
    };

    const intlProviderProps = {
        defaultLocale: 'en',
        locale: 'en',
    };

    const renderComponent = (props = {}) => {
        return render(
            <IntlProvider {...intlProviderProps}>
                <Provider store={store}>
                    <CustomEnableDisableGuestAccountsSetting
                        {...baseProps}
                        {...props}
                    />
                </Provider>
            </IntlProvider>);
    };

    const warningMessage = (
        <FormattedMessage
            defaultMessage='All current guest account sessions will be revoked, and marked as inactive'
            id='admin.guest_access.disableConfirmWarning'
        />
    );

    describe('renders correctly', () => {
        test('when enabled', () => {
            const wrapper = renderComponent({value: true});
            expect(wrapper).toMatchSnapshot();
        });

        test('when disabled', () => {
            const wrapper = renderComponent({value: false});
            expect(wrapper).toMatchSnapshot();
        });
    });

    describe('handleChange', () => {
        test('should enable without show confirmation modal or warning', () => {
            const props = {
                showConfirm: true,
                onChange: jest.fn(),
            };

            renderComponent(props);

            const trueRadio = screen.getByTestId('MySettingtrue');
            fireEvent.click(trueRadio);

            expect(props.onChange).toHaveBeenCalledWith(baseProps.id, true, false, false, '');
        });

        test('should show confirmation modal and warning when disabling', () => {
            const props = {
                value: true,
                showConfirm: true,
                onChange: jest.fn(),
            };

            renderComponent(props);

            const falseRadio = screen.getByTestId('MySettingfalse');
            fireEvent.click(falseRadio);

            expect(props.onChange).toHaveBeenCalledWith(baseProps.id, false, true, false, warningMessage);
        });

        test('should call onChange with doSubmit = true when confirm is true', () => {
            const props = {
                ...baseProps,
                onChange: jest.fn(),
                showConfirm: true,
            };

            renderComponent(props);

            const falseRadio = screen.getByTestId('MySettingfalse');
            fireEvent.click(falseRadio);

            const confirmButton = screen.getByText('Save and Disable Guest Access');
            fireEvent.click(confirmButton);

            expect(props.onChange).toHaveBeenCalledWith(baseProps.id, false, true, true, warningMessage);
        });
    });
});
