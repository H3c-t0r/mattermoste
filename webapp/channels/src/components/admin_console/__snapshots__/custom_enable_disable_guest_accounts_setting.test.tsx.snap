// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/AdminConsole/CustomEnableDisableGuestAccountsSetting initial state with false 1`] = `
<Fragment>
  <BooleanSetting
    disabled={false}
    falseText={
      <Memo(MemoizedFormattedMessage)
        defaultMessage="False"
        id="admin.false"
      />
    }
    helpText={
      <FormattedMarkdownMessage
        defaultMessage="When true, external guest can be invited to channels within teams. Please see [Permissions Schemes](../user_management/permissions/system_scheme) for which roles can invite guests."
        id="admin.guest_access.enableDescription"
      />
    }
    id="MySetting"
    label={
      <Memo(MemoizedFormattedMessage)
        defaultMessage="Enable Guest Access: "
        id="admin.guest_access.enableTitle"
      />
    }
    onChange={[Function]}
    setByEnv={false}
    trueText={
      <Memo(MemoizedFormattedMessage)
        defaultMessage="True"
        id="admin.true"
      />
    }
    value={false}
  />
  <ConfirmModal
    confirmButtonClass="btn btn-primary"
    confirmButtonText={
      <Memo(MemoizedFormattedMessage)
        defaultMessage="Save and Disable Guest Access"
        id="admin.guest_access.disableConfirmButton"
      />
    }
    message={
      <Memo(MemoizedFormattedMessage)
        defaultMessage="Disabling guest access will revoke all current Guest Account sessions. Guests will no longer be able to login and new guests cannot be invited into Mattermost. Guest users will be marked as inactive in user lists. Enabling this feature will not reinstate previous guest accounts. Are you sure you wish to remove these users?"
        id="admin.guest_access.disableConfirmMessage"
      />
    }
    modalClass=""
    onCancel={[MockFunction]}
    onConfirm={[Function]}
    show={false}
    title={
      <Memo(MemoizedFormattedMessage)
        defaultMessage="Save and Disable Guest Access?"
        id="admin.guest_access.disableConfirmTitle"
      />
    }
  />
</Fragment>
`;

exports[`components/AdminConsole/CustomEnableDisableGuestAccountsSetting initial state with true 1`] = `
<Fragment>
  <BooleanSetting
    disabled={false}
    falseText={
      <Memo(MemoizedFormattedMessage)
        defaultMessage="False"
        id="admin.false"
      />
    }
    helpText={
      <FormattedMarkdownMessage
        defaultMessage="When true, external guest can be invited to channels within teams. Please see [Permissions Schemes](../user_management/permissions/system_scheme) for which roles can invite guests."
        id="admin.guest_access.enableDescription"
      />
    }
    id="MySetting"
    label={
      <Memo(MemoizedFormattedMessage)
        defaultMessage="Enable Guest Access: "
        id="admin.guest_access.enableTitle"
      />
    }
    onChange={[Function]}
    setByEnv={false}
    trueText={
      <Memo(MemoizedFormattedMessage)
        defaultMessage="True"
        id="admin.true"
      />
    }
    value={true}
  />
  <ConfirmModal
    confirmButtonClass="btn btn-primary"
    confirmButtonText={
      <Memo(MemoizedFormattedMessage)
        defaultMessage="Save and Disable Guest Access"
        id="admin.guest_access.disableConfirmButton"
      />
    }
    message={
      <Memo(MemoizedFormattedMessage)
        defaultMessage="Disabling guest access will revoke all current Guest Account sessions. Guests will no longer be able to login and new guests cannot be invited into Mattermost. Guest users will be marked as inactive in user lists. Enabling this feature will not reinstate previous guest accounts. Are you sure you wish to remove these users?"
        id="admin.guest_access.disableConfirmMessage"
      />
    }
    modalClass=""
    onCancel={[MockFunction]}
    onConfirm={[Function]}
    show={false}
    title={
      <Memo(MemoizedFormattedMessage)
        defaultMessage="Save and Disable Guest Access?"
        id="admin.guest_access.disableConfirmTitle"
      />
    }
  />
</Fragment>
`;

exports[`components/AdminConsole/CustomEnableDisableGuestAccountsSetting renders correctly when disabled 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="form-group"
        data-testid="MySetting"
      >
        <label
          class="control-label col-sm-4"
          for="MySetting"
        >
          Enable Guest Access: 
        </label>
        <div
          class="col-sm-8"
        >
          <a
            id="MySetting"
          />
          <label
            class="Label-eIDBis glIJuw"
          >
            <input
              data-testid="MySettingtrue"
              id="MySettingtrue"
              name="MySetting"
              type="radio"
              value="true"
            />
            True
          </label>
          <label
            class="Label-eIDBis glIJuw"
          >
            <input
              checked=""
              data-testid="MySettingfalse"
              id="MySettingfalse"
              name="MySetting"
              type="radio"
              value="false"
            />
            False
          </label>
          <div
            class="help-text"
            data-testid="MySettinghelp-text"
          >
            <span>
              When true, external guest can be invited to channels within teams. Please see 
              <a
                href="../user_management/permissions/system_scheme"
                rel="noopener noreferrer"
                target="_blank"
              >
                Permissions Schemes
              </a>
               for which roles can invite guests.
            </span>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="form-group"
      data-testid="MySetting"
    >
      <label
        class="control-label col-sm-4"
        for="MySetting"
      >
        Enable Guest Access: 
      </label>
      <div
        class="col-sm-8"
      >
        <a
          id="MySetting"
        />
        <label
          class="Label-eIDBis glIJuw"
        >
          <input
            data-testid="MySettingtrue"
            id="MySettingtrue"
            name="MySetting"
            type="radio"
            value="true"
          />
          True
        </label>
        <label
          class="Label-eIDBis glIJuw"
        >
          <input
            checked=""
            data-testid="MySettingfalse"
            id="MySettingfalse"
            name="MySetting"
            type="radio"
            value="false"
          />
          False
        </label>
        <div
          class="help-text"
          data-testid="MySettinghelp-text"
        >
          <span>
            When true, external guest can be invited to channels within teams. Please see 
            <a
              href="../user_management/permissions/system_scheme"
              rel="noopener noreferrer"
              target="_blank"
            >
              Permissions Schemes
            </a>
             for which roles can invite guests.
          </span>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`components/AdminConsole/CustomEnableDisableGuestAccountsSetting renders correctly when enabled 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="form-group"
        data-testid="MySetting"
      >
        <label
          class="control-label col-sm-4"
          for="MySetting"
        >
          Enable Guest Access: 
        </label>
        <div
          class="col-sm-8"
        >
          <a
            id="MySetting"
          />
          <label
            class="Label-eIDBis glIJuw"
          >
            <input
              checked=""
              data-testid="MySettingtrue"
              id="MySettingtrue"
              name="MySetting"
              type="radio"
              value="true"
            />
            True
          </label>
          <label
            class="Label-eIDBis glIJuw"
          >
            <input
              data-testid="MySettingfalse"
              id="MySettingfalse"
              name="MySetting"
              type="radio"
              value="false"
            />
            False
          </label>
          <div
            class="help-text"
            data-testid="MySettinghelp-text"
          >
            <span>
              When true, external guest can be invited to channels within teams. Please see 
              <a
                href="../user_management/permissions/system_scheme"
                rel="noopener noreferrer"
                target="_blank"
              >
                Permissions Schemes
              </a>
               for which roles can invite guests.
            </span>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="form-group"
      data-testid="MySetting"
    >
      <label
        class="control-label col-sm-4"
        for="MySetting"
      >
        Enable Guest Access: 
      </label>
      <div
        class="col-sm-8"
      >
        <a
          id="MySetting"
        />
        <label
          class="Label-eIDBis glIJuw"
        >
          <input
            checked=""
            data-testid="MySettingtrue"
            id="MySettingtrue"
            name="MySetting"
            type="radio"
            value="true"
          />
          True
        </label>
        <label
          class="Label-eIDBis glIJuw"
        >
          <input
            data-testid="MySettingfalse"
            id="MySettingfalse"
            name="MySetting"
            type="radio"
            value="false"
          />
          False
        </label>
        <div
          class="help-text"
          data-testid="MySettinghelp-text"
        >
          <span>
            When true, external guest can be invited to channels within teams. Please see 
            <a
              href="../user_management/permissions/system_scheme"
              rel="noopener noreferrer"
              target="_blank"
            >
              Permissions Schemes
            </a>
             for which roles can invite guests.
          </span>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
